valhalla/mjolnir/landmark_builder.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <boost/property_tree/ptree.hpp>
+#include <string>
+#include <unordered_map>
+
+namespace valhalla {
+namespace mjolnir {
+static const std::string default_landmark_name = "";
+
+// obvious landmark types for vehicle routing
+enum class LandmarkType : uint8_t {
+  null = 0,
+  restaurant = 1,
+  school = 2,
+  fast_food = 3,
+  cafe = 4,
+  fuel = 5,
+  bank = 6,
+  pharmacy = 7,
+  kindergarten = 8,
+  bar = 9,
+  hospital = 10,
+  post_office = 11,
+  pub = 12,
+  clinic = 13,
+  police = 14,
+  fire_station = 15,
+  car_wash = 16,
+  charging_station = 17,
+  college = 18,
+  university = 19,
+  theatre = 20,
+  cinema = 21
+};
+
+using Landmark = std::tuple<std::string, LandmarkType, double, double>;
+
+inline LandmarkType string_to_landmark_type(const std::string& s) {
+  static const std::unordered_map<std::string, LandmarkType> string_to_landmark_type =
+      {{"restaurant", LandmarkType::restaurant},
+       {"school", LandmarkType::school},
+       {"fast_food", LandmarkType::fast_food},
+       {"cafe", LandmarkType::cafe},
+       {"fuel", LandmarkType::fuel},
+       {"bank", LandmarkType::bank},
+       {"pharmacy", LandmarkType::pharmacy},
+       {"kindergarten", LandmarkType::kindergarten},
+       {"bar", LandmarkType::bar},
+       {"hospital", LandmarkType::hospital},
+       {"post_office", LandmarkType::post_office},
+       {"pub", LandmarkType::pub},
+       {"clinic", LandmarkType::clinic},
+       {"police", LandmarkType::police},
+       {"fire_station", LandmarkType::fire_station},
+       {"car_wash", LandmarkType::car_wash},
+       {"charging_station", LandmarkType::charging_station},
+       {"college", LandmarkType::college},
+       {"university", LandmarkType::university},
+       {"theatre", LandmarkType::theatre},
+       {"cinema", LandmarkType::cinema}};
+
+  auto it = string_to_landmark_type.find(s);
+  if (it != string_to_landmark_type.cend()) {
+    return it->second;
+  }
+  return LandmarkType::null;
```

Comment:
we can just disallow conversion when its unknown
```suggestion
  if (it == string_to_landmark_type.cend())
    throw std::runtime_error("unknown landmark type");
  return it->second;
```

Reply:
I just realize this function is called in `node_callback` to check every node with amenity. There are many amenity values not included in the list so we shouldn't throw. WDYT? @kevinkreiser if we remove LandmarkType::null how to do the check?

Reply:
ah we can use try catch

Reply:
exactly